aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -52,13 +52,49 @@
 
 namespace at::native {
 
+
+template <typename args_t, size_t... Is>
+constexpr auto sum_of_sizes(args_t args, std::index_sequence<Is...>) {
+    if constexpr (sizeof...(Is) == 0) {
+      return 0;
+    } else {
+      return (sizeof(std::tuple_element_t<Is, args_t>) + ...);
+    }
+}
+
+template <int io_sizes>
+constexpr auto elems_per_thread(){
+  if constexpr (io_sizes == 1) {
+    return 16;
+  } else if constexpr (io_sizes < 4) {
+    return 8;
+  } else {
+    return 4;
+  }
```

Comment:
nit: can this be expressed as something like `bytes_per_thread//size`?
Just thinking about this potentially being potentially hw-dependent where we might want different bytes per thread

Reply:
For ease of reasoning it's better if it's power of 2, and `bytest_per_thread//size` won't have this property. Besides, for large sizes we still probably want to keep 4, it served us well all these years.